### ISSUE TITLE
Table name prefix (and suffix) are not entirely working

### ISSUE THREAD

**Issue by walterdavis** [scenic-views/scenic#295]
If your application has configured a table name prefix, using the usual `config.active_record.table_name_prefix = 'feds_api_'` method in application.rb, Scenic will properly resolve that name throughout its chain, except when it comes time to locate the SQL file containing the view query.

```sh
No such file or directory @ rb_sysopen - /home/waldavis/projects/feds-api/db/views/feds_api_foos_v01.sql
```

This is an accurate warning, because the generated SQL file from `rails g scenic:view foo` is really named `foos_v01.sql`.

It's important to note that whenever prefixes are used in Rails, you never refer to them in the migration tooling at all. That is to say, if your prefix is `foo_` and you want to create a migration for `bars`, you just use the command `rails g migration create_bars field, field2, field3...` and the resulting migration will not refer to `foo_bars` internally, either, but the generated table will be named `foo_bars`.

So back to our prefixes and Scenic -- this really gets ugly if you naively fix the problem by renaming the generated SQL file so the migration can find it. This works, but only for version 1. If you follow the directions to update, which is to issue the same command you used to create the first (or Nth) version: `rails g scenic:view foo`. Now you get the error that you are going to replace the original migration. Since Scenic uses the names of the files to derive its versioning, it thinks you are trying to create the view again for the first time, and not finding any `foos_v01.sql` it naturally tries to create one, along with the initial `[timestamp]_create_foos.rb`, which already exists.

Finally, if you try to work around the system for the upgrades, by using the command `rails g scenic:view feds_api_foo`, then everything works as documented, except that the generated migration file reads `update_view :feds_api_foo, version: 2, revert_to_version: 1` internally, and then fails to run, complaining that it cannot find a file named `feds_api_feds_api_foos_v02.sql` in the views directory. Again, this is accurate, as the file is named with only one instance of the prefix.

It seems to me that the simplest possible solution would be to ensure that the file path resolution during the migration ignores the prefix. I've fought with this problem in another gem before, and I know that "simple" is a very loaded word here. You're either going to need to do that, or ensure that within your gem, the files are written out with their prefix attached. Either one of those will get around the problem.

Everything I've written here applies to the suffix as well as the prefix.

**Comment by derekprior** [comment]
This seems somewhat tricky. It _sounds_ like the only place scenic should care about the prefix and suffix is when it goes to actually issue the `create view` statement where we should make sure the name is prefixed and suffixed correctly. The file names and the name as it appears in migrations should all be unprefixed. Is that your understanding?

Initially, I thought we could just check `ActiveRecord::Base.table_name_prefix` and `ActiveRecord::Base.table_name_suffix` when we go to operate on the views in the database only, but it seems we may also need to remove those prefix and suffixes when we dump the views. When dumping tables that use a prefix, are the table names in `schema.rb` unprefixed?

An additional concern of mine is that the prefix and suffix could actually be set _anywhere_, right? You don't have to set them on `ActiveRecord::Base`, you could instead set them on the model, on `ApplicationRecord`, or anywhere else in the hierarchy tree. I guess we would just need to figure out how rails handles this ambiguity and replicate that?

**Comment by walterdavis** [comment]
That sounds about right.

**Comment by derekprior** [comment]
Whelp this was an adventure. When you have a simple migration like this:

```ruby
def change
  create_view :searches
end
```

By the time Scenic's `create_view` method is called, `:searches` has already been mutated to contain the prefix and suffix if appropriate. You can see it happen here:

https://github.com/rails/rails/blob/4607108f91881cd1c24285dca63dc8e0f3f8a4f1/activerecord/lib/active_record/migration.rb#L917-L933

That's why the migration looks for the prefixed filename in the first place. Off the top of my head, that means we'd have a couple of potential options:

1. Reverse `proper_table_name` in Scenic so we can access the unprefixed name and use that for file lookup
2. Update the generators to create prefixed SQL files so things "just work" (at least on creation -- I haven't tested dumping the views yet)

I'm not really sure which is better... 😕

**Comment by derekprior** [comment]
@walterdavis Proposed fix in #344 if you want to give it a whirl
